src/api/endpoints/auth/TokenManager.ts

```diff
@@ -32,19 +32,20 @@ export default class TokenManager extends ApiEndpoint {
 
   public async getAccessToken(source: string = "unknown"): Promise<string | undefined> {
     this.log?.debug(`getAccessToken: New token requested from ${source}`);
-    if (this.accessToken !== undefined && this.accessTokenExpiration !== undefined) {
+    if ((this.accessToken !== undefined || this.config.cookieAuth) && this.accessTokenExpiration !== undefined) {
```

Comment:
Do we want to enter here if accessToken is potentially undefined?

Reply:
Yes because if this.config.cookieAuth is true then this.accessToken is undefined but we still want to check the accessTokenExpiration to know if we need to ask the API to refresh it by setting new cookies